src/Plugin/MessagePurge/Days.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Drupal\message\Plugin\MessagePurge;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\message\MessagePurgeBase;
+use Drupal\message\MessageTemplateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Delete messages older than certain days.
+ *
+ * @MessagePurge(
+ *  id = "days",
+ *  label = @Translation("Days", context = "MessagePurge"),
+ *  description = @Translation("Delete messages older than a given amount of days."),
+ * )
+ */
+class Days extends MessagePurgeBase {
+
+  /**
+   * The request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * Days constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration.
+   * @param string $plugin_id
+   *   The plugin ID.
+   * @param mixed $plugin_definition
+   *   The plugin definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\Query\QueryInterface $message_query
+   *   The entity query service.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   *   The request stack used to determine the current time.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, QueryInterface $message_query, RequestStack $request_stack) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $message_query);
+    $this->requestStack = $request_stack;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.query')->get('message'),
+      $container->get('request_stack')
+    );
+  }
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['days'] = [
+      '#type' => 'number',
+      '#min' => 1,
+      '#title' => t('Messages older than'),
+      '#description' => t('Maximal message age in days.'),
+      '#default_value' => $this->configuration['days'],
+      '#tree' => FALSE,
```

Comment:
No, but it's set to TRUE for the each plugin, so must be returned to false so the plugin form handler need not be aware of the exterior tree structure.